mutable_buffer/src/persistence_windows.rs

```diff
@@ -0,0 +1,602 @@
+
+        let c = &w.closed[1];
+        assert_eq!(c.row_count, 2);
+        assert_eq!(c.min_time, third_start);
+        assert_eq!(c.max_time, third_end);
+        assert_eq!(c.created_at, third_created_at);
+    }
+
+    #[test]
+    fn flush_persistable_overlaps_open() {
+        let mut w = PersistenceWindows::new(2);
+        let created_at = Instant::now();
+        let start_time = Utc::now();
+        std::thread::sleep(Duration::new(0, 1_000));
+        let third_start = Utc::now();
+        std::thread::sleep(Duration::new(0, 1_000));
+
+        let first_end = Utc::now();
+        w.add_range(1, 2, 2, start_time, first_end, created_at);
+
+        std::thread::sleep(Duration::new(0, 1_000));
+        let second_start = Utc::now();
+        let second_created_at = created_at
+            .checked_add(Duration::new(CLOSED_WINDOW_SECONDS as u64 * 3, 0))
+            .unwrap();
+        std::thread::sleep(Duration::new(0, 1_000));
+        let second_end = Utc::now();
+        w.add_range(1, 3, 3, second_start, second_end, second_created_at);
+
+        let third_created_at = second_created_at
+            .checked_add(Duration::new(CLOSED_WINDOW_SECONDS as u64, 0))
+            .unwrap();
+        let third_end = Utc::now();
+        w.add_range(1, 5, 2, third_start, third_end, third_created_at);
+
+        let end_at = third_created_at.checked_add(Duration::new(1, 0)).unwrap();
+        w.rotate(end_at);
+
+        let t_flush = w.t_flush().unwrap();
+        assert_eq!(t_flush, first_end);
+        let c = w.persistable.as_ref().unwrap();
+        assert_eq!(c.created_at, created_at);
+        assert_eq!(c.row_count, 2);
+        assert_eq!(c.min_time, start_time);
+        assert_eq!(c.max_time, first_end);
+
+        let mins = w.persistable.as_ref().unwrap().sequencer_numbers.clone();
+        assert_eq!(mins, w.minimum_unpersisted_sequence().unwrap());
+        w.flush();
+        assert!(w.persistable.is_none());
+        let mins = w.closed[0].sequencer_numbers.clone();
+        assert_eq!(mins, w.minimum_unpersisted_sequence().unwrap());
+
+        // the closed window should have a min time equal to the flush
+        let c = &w.closed[0];
+        assert_eq!(c.row_count, 3);
+        assert_eq!(c.min_time, second_start);
+        assert_eq!(c.max_time, second_end);
+        assert_eq!(c.created_at, second_created_at);
+
+        // the open window should have a min time equal to t_flush
+        let c = w.open.as_ref().unwrap();
+        assert_eq!(c.row_count, 2);
+        assert_eq!(c.min_time, t_flush);
+        assert_eq!(c.max_time, third_end);
+        assert_eq!(c.created_at, third_created_at);
+    }
+
+    #[test]
+    fn flush_persistable_overlaps_open_and_closed() {
+        let mut w = PersistenceWindows::new(2);
+        let created_at = Instant::now();
+        let start_time = Utc::now();
+        std::thread::sleep(Duration::new(0, 1_000));
+        let second_start = Utc::now();
```

Comment:
```suggestion
        let second_start = start_time + chrono::Duration::microseconds(1);
```

No need to actually sleep here.